From 08f8bb1431e1536b07118e0fe85826d7958b56a7 Mon Sep 17 00:00:00 2001
From: Hugo Mano <hugo@zml.ai>
Date: Tue, 27 May 2025 11:50:40 +0200
Subject: [PATCH 3/6] Expose nvptxcompiler to link against in XLA if
 enable_libnvptxcompiler_support is set

Only for ZML, no PR on XLA side.
---
 third_party/gpus/cuda/hermetic/cuda_nvcc.BUILD.tpl |  8 ++++++++
 xla/stream_executor/cuda/BUILD                     | 12 +++++++++++-
 2 files changed, 19 insertions(+), 1 deletion(-)

diff --git a/third_party/gpus/cuda/hermetic/cuda_nvcc.BUILD.tpl b/third_party/gpus/cuda/hermetic/cuda_nvcc.BUILD.tpl
index 733d95d371..f6c1c5ab53 100644
--- a/third_party/gpus/cuda/hermetic/cuda_nvcc.BUILD.tpl
+++ b/third_party/gpus/cuda/hermetic/cuda_nvcc.BUILD.tpl
@@ -52,6 +52,14 @@ filegroup(
     visibility = ["//visibility:public"],
 )
 
+filegroup(
+    name = "libnvptxcompiler",
+    srcs = [
+        "lib/libnvptxcompiler_static.a",
+    ],
+    visibility = ["//visibility:public"],
+)
+
 filegroup(
     name = "bin",
     srcs = glob([
diff --git a/xla/stream_executor/cuda/BUILD b/xla/stream_executor/cuda/BUILD
index 4a62a44357..d5f723fd62 100644
--- a/xla/stream_executor/cuda/BUILD
+++ b/xla/stream_executor/cuda/BUILD
@@ -84,6 +84,11 @@ config_setting(
     },
 )
 
+cc_import(
+    name = "nvptxcompiler",
+    static_library = "@cuda_nvcc//:libnvptxcompiler",
+)
+
 cc_library(
     name = "cuda_platform_id",
     srcs = ["cuda_platform_id.cc"],
@@ -128,7 +133,12 @@ cc_library(
             "@tsl//tsl/platform:errors",
             "@tsl//tsl/platform:status",
             "@tsl//tsl/platform:statusor",
-        ] + tf_additional_cuda_platform_deps(),
+        ] + tf_additional_cuda_platform_deps() + select({
+            ":libnvptxcompiler_support_enabled": [
+                ":nvptxcompiler",
+            ],
+            "//conditions:default": [],
+        }),
     alwayslink = True,  # Registers itself with the PlatformManager.
 )
 
-- 
2.39.5 (Apple Git-154)

